poetry/utils/setup_reader.py

```diff
@@ -1,379 +1,96 @@
-        result["extras_require"] = self._find_extras_require(setup_call, body)
-        result["python_requires"] = self._find_single_string(
-            setup_call, body, "python_requires"
-        )
+        with pep517.envbuild.BuildEnvironment() as env, temporary_directory() as tmp_dir:
+            env.pip_install([cls.build_requires])
+            reqs = hooks.get_requires_for_build_wheel({})
+            env.pip_install(reqs)
+
+            dist_info = hooks.prepare_metadata_for_build_wheel(tmp_dir)
+            distribution = PathDistribution(Path(tmp_dir) / dist_info)
+
+            result = {
+                "name": distribution.metadata["Name"],
+                "version": distribution.version,
+                "summary": distribution.metadata.get("Summary"),
+                "install_requires": [],
+                "python_requires": distribution.metadata.get("Requires-Python"),
+                "extras_require": {},
+            }
+
+            if distribution.requires:
+                for record in distribution.requires:
+                    requirements = record.split(";", 1)
+                    project_name = (
+                        requirements[0]
+                        .replace("(", "")
+                        .replace(")", "")
+                        .replace(" ", "")
+                    )
+
+                    try:
+                        marker = requirements[1]
+                        has_extra = re.search(r".*\bextra == ['\"](.*)['\"]", marker)
+
+                        if has_extra:
+                            extra_group = has_extra.groups()[0]
+                            marker = re.sub(
+                                r"(and |or )?extra == ['\"]{extra_group}['\"]".format(
+                                    extra_group=extra_group
+                                ),
+                                "",
+                                marker,
+                            ).strip()
+
+                            if marker:
+                                extra_group += ":" + marker.strip(" ()")
+
+                            if extra_group in result["extras_require"]:
+                                result["extras_require"][extra_group].append(
+                                    project_name
+                                )
+                            else:
+                                result["extras_require"][extra_group] = [project_name]
+                        else:
+                            result["install_requires"].append(
+                                "{project_name};{marker}".format(
+                                    project_name=project_name, marker=marker
+                                )
+                            )
+
+                    except IndexError:
+                        result["install_requires"].append(project_name)
+
+        egg_info = Path(directory).glob("*.egg-info")
```

Comment:
We should not need this anymore correct? Since the metadata will be generated in the isolated environment?